lit_llama/adapter.py

```diff
@@ -235,9 +295,9 @@ def forward(
                     x, rope, mask, max_seq_length, input_pos, self.kv_caches[i], self.adapter_kv_caches[i]
                 )
 
-        x = self.transformer.ln_f(x)
+        x = self.transformer.ln_f(x) # (B, T, n_embd)
 
-        logits = self.lm_head(x)  # (b, t, vocab_size)
+        logits = self.lm_head(x)  # (B, T, vocab_size)
```

Comment:
I think there's a bug in that the lm_head definition and wte should use the `padded_vocab_size` as is done for the base model

Reply:
Again, I agree. But this annotation is correct for the current state of the code. So I propose to keep it as is for now and create a separate PR with this fix and the ones above.
I can do this right after this PR is merged.

Reply:
While we are discussing `lm_head` what about [weight tying](https://paperswithcode.com/method/weight-tying) like in [nanoGPT code](https://github.com/karpathy/nanoGPT/blob/master/model.py#L145)?

I assume you don't use it because in the [LLaMa repo](https://github.com/facebookresearch/llama/blob/main/llama/model.py#L205-L214) it's not implemented and thus might be a problem with loading pretrained weights. But maybe there is another reason for it. 

Asking just of curiosity 🤓 .